src/codegen/expression.rs

```diff
@@ -2552,12 +2584,25 @@ fn array_subscript(
                         array_length
                     }
                 } else {
-                    Expression::Builtin(
+                    //if a subscript is encountered array length will be called
+
+                    //return array length by default
+                    let mut returned = Expression::Builtin(
```

Comment:
There is no need to instantiate this variable. Create the `Expression::Builtin` as the last statement of this else-branch.